daemon/delete.go

```diff
@@ -79,7 +79,12 @@ func (daemon *Daemon) rmLink(container *container.Container, name string) error
 func (daemon *Daemon) cleanupContainer(container *container.Container, forceRemove, removeVolume bool) (err error) {
 	if container.IsRunning() {
 		if !forceRemove {
-			err := fmt.Errorf("You cannot remove a running container %s. Stop the container before attempting removal or use -f", container.ID)
+			state := container.StateString()
+			procedure := "Stop the container before attempting removal or use -f"
```

Comment:
OK sounds good to me.
Just few points I need to ask:
 * Can I add the state and the container ID in the response headers?
* To make this backward compatible, can I keep the error message as it is?
* In the client I will read the state and the container ID headers, if they are empty the error message sent by the daemon will be returned. Otherwise, the client will return a generated error message using the above headers.

WDYT?